queries.sql

```diff
@@ -39,6 +39,31 @@ SELECT AVG(weight_kg) FROM animals;
+'Dean Winchester' AND escape_attempts > 0;
+SELECT full_name, MAX(owner_count) FROM (SELECT full_name, COUNT(owner_id) AS owner_count FROM animals a JOIN owners o ON a.owner_id = o.id GROUP BY full_name, owner_id) AS subquery GROUP BY full_name;
 
+/* Vet clinic database: add "join table" for visits */
 
-
+SELECT animals.name, vets.name, MAX(date_of_visit) AS last_vist FROM visits JOIN animals ON visits.animals_id = animals.id JOIN vets ON visits.vets_id = vets.id WHERE vets.name = 'William Tatcher' GROUP BY animals.name, vets.na
+me ORDER BY last_vist DESC LIMIT 1;
+SELECT animals.name, vets.name, COUNT(visits.animals_id) FROM visits JOIN animals ON visits.animals_id = animals.id JOIN vets ON vets.id = visits.vets_id WHERE vets.id = 3 GROUP BY animals.name, vets.name;
+SELECT vets.name, species.name, specializations.species_id FROM SPECIALIZATIONS FULL JOIN vets ON specializations.vets_id = vets.id FULL JOIN species ON species.id = specializations. species_id GROUP BY specializati
+ons.species_id, vets.name, species.name;
+SELECT animals.name AS animals_that_visited, visits.date_of_visit, vets.name FROM VISITS JOIN ANIMALS ON animals.id = visits.animals_id JOIN vets ON vets.id = visits.vets_id WHERE vets.id = 3 AND date_of_visit BETWEEN '2020-04-1' AND '2020-08-30' GROUP BY animals.name, vets.name, visits.date_of_visit;
+SELECT animals.name, COUNT(visits.animals_id) AS Animal_visits FROM visits JOIN animals ON animals.id = animals_id JOIN vets ON vets.id = visits.vets_id GROUP BY animals.name ORDER BY Animal_visits desc;
+SELECT animals.name, COUNT(visits.animals_id) AS Animal_visits FROM visits JOIN animals ON animals.id = animals_id JOIN vets ON vets.id = visits.vets_id GROUP BY animals.name ORDER BY Animal_visits;
+vet_clinic=# SELECT animals.name, COUNT(visits.animals_id) AS Animal_visits FROM visits JOIN animals ON animals.id = animals_id JOIN vets ON vets.id = visits.vets_id GROUP BY animals.name ORDER BY Animal_visits desc limit 2;
+SELECT animals.name AS animal_name, vets.name, MIN(visits.date_of_visit) FROM VISITS JOIN animals ON animals.id = visits.animals_id JOIN vets ON vets.id = visits.vets_id WHERE vets.name = 'Maisy Smith' group by animals.name, vets.name, visits.date_of_visit limit 1;
+SELECT * FROM visits JOIN animals ON visits.animals_id = animals.id JOIN vets ON vets.id = visits.vets_id ORDER BY visits.date_of_visit desc limit 1;
+SELECT COUNT(*) FROM visits JOIN (SELECT vets.id FROM vets FULL JOIN specializations ON vets.id = specializations.vets_id FULL JOIN species ON species.id = specializations.species_id WHERE specializations.species_id IS NULL) AS
+ vet ON vet.id = visits.vets_id;
+ SELECT s.name AS species_name, COUNT(vis.date_of_visit) AS total_visits FROM visits AS vis JOIN animals AS 
+a ON vis.animals_id = a.id JOIN species AS s ON a.species_id = s.id JOIN vets AS v ON vis.vets_id = v.id WHERE v.name = 
+ 'Maisy Smith' GROUP BY s.name ORDER BY total_visits DESC LIMIT 1;
```

Comment:
- [ ] Kindly add screenshots of the results you got after running all queries to your PR. The screenshots will determine if you successfully ran all the queries and if the queries are correct.